app/src/main/java/com/dsu/poc/ui/login/LoginFragment.kt

```diff
@@ -1,6 +1,47 @@
 package com.dsu.poc.project.ui.login
 
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.dsu.poc.R
 import com.dsu.poc.base.BaseFragment
+import com.dsu.poc.databinding.LoginFragmentBinding
+
+class LoginFragment : BaseFragment(){  lateinit var loginView: View
+    lateinit var loginViewModel: LoginViewModel
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        loginViewModel = ViewModelProviders.of(activity!!).get(LoginViewModel::class.java)
+    }
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val loginBinder : LoginFragmentBinding = DataBindingUtil.inflate(inflater,
+            R.layout.main_login,container,false)
+        loginView = loginBinder.getRoot()
+        loginBinder.loginbtn.setOnClickListener(View.OnClickListener {
+            val email:String = loginBinder.editUserName.getText().toString()
+            val password:String = loginBinder.editPassword.text.toString()
+            loginViewModel.validateCredentials(email,password).observe(this,object:
+                Observer<String> {
+                override fun onChanged(t: String?) {
```

Comment:
what is the necessity of overloading onChanged function? doesn't this counter the concept of live data and observer pattern.